### PR TITLE
implementation of end() method

### PR DESCRIPTION
**Pull request by lyusupov**
Fix for https://github.com/adafruit/Adafruit_SPIFlash/issues/73

### CODE REVIEW

File: src/qspi/Adafruit_FlashTransport_QSPI_SAMD.cpp [hunk 1]
```diff
@@ -80,6 +80,12 @@ void Adafruit_FlashTransport_QSPI::begin(void) {
   QSPI->CTRLA.bit.ENABLE = 1;
 }
 
+void Adafruit_FlashTransport_QSPI::end(void) {
+  QSPI->CTRLA.bit.ENABLE = 0;
+
+  /* NEEDS WORK */
```
Comment (hathach) [inline src/qspi/Adafruit_FlashTransport_QSPI_SAMD.cpp:86]: We should also disable the qspi clock as well (reversed to being() ) to save power. 

```C
MCLK->APBCMASK.bit.QSPI_ = false;
MCLK->AHBMASK.bit.QSPI_ = false;
MCLK->AHBMASK.bit.QSPI_2X_ = false;
```

**Review by hathach — COMMENTED** [review]
Superb! everything changes look perfect. Only one request to also disable the QSPI clock when end(), then we are good to merge.

File: src/qspi/Adafruit_FlashTransport_QSPI_SAMD.cpp (hunk 1 above)
Comment (lyusupov) [reply src/qspi/Adafruit_FlashTransport_QSPI_SAMD.cpp]: Done in https://github.com/adafruit/Adafruit_SPIFlash/pull/108/commits/a40a06418d01698968022b08f45d7cec3bf184b7

**Review by hathach — APPROVED** [review]
perfect, good to merge now.

**Pull request merged**